DOCSP-30343: Move RealmSet content to CRUD pages

cbullinger opened the pull request:
## Pull Request Info

Move RealmSet content from [Data Types](https://www.mongodb.com/docs/realm/sdk/kotlin/realm-database/schemas/supported-types/) page to respective CRUD pages

### Jira

- https://jira.mongodb.org/browse/DOCSP-30343

### Staged Changes

- [Data Types](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/schemas/supported-types/#realmset)
- [Define Object Model](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/schemas/define-realm-object-model/#define-a-realmset)
- [Create](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/crud/create/)
- [Read](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/crud/read/)
- [Update](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/crud/update/)
- [Delete](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-30343-create/sdk/kotlin/realm-database/crud/delete/)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [x] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

dacharyc commented on source/sdk/kotlin/realm-database/crud/read.txt:
```diff
@@ -71,6 +71,44 @@ as you would a Kotlin Map.
+Then, we check if the ``RealmSet`` contains this ``Snack`` object
+by passing the read-only set to ``set.contains()``:
+
+.. literalinclude:: /examples/generated/kotlin/SchemaTest.snippet.set-contains.kt
```
When these snippets were with the Create example, it was easy to just scroll a little to see what a `Snack` was and see us querying for a `Frog` named "Kermit". Now with these split up, we've lost the context of the Frog query or knowing what these other code elements are (`Snack`) that we're referring to in the snippet. Also, the text says that we're passing a `Snack` to `setOf()` and passing that set to `set.contains()` but we're only showing the `set.contains()` part.

Do you think we should expand these snippets to show querying for Kermit, finding or creating the set element we want to query for, and performing the `set.contains()` query? Or do you think that just adds unnecessary verbosity to the example?

(Same Q for down below.)

dacharyc commented on source/sdk/kotlin/realm-database/crud/delete.txt:
```diff
@@ -134,3 +134,31 @@ Deleting the parent object automatically deletes all of its embedded objects.
+delete to 
+`set.remove() <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-set/index.html#-1503494415%2FFunctions%2F-1651551339>`__: 
+
+.. literalinclude:: /examples/generated/kotlin/SchemaTest.snippet.remove-item-from-set.kt
```
Same Q here as in the Read section - do we want to show the context of getting the object that has the set we're removing elements from, or do you think that adds unnecessary verbosity to the example?

dacharyc reviewed (approved):
I have some Qs about whether we want to expand examples to show more of the code now that they're splitting up, but they're non-blocking Qs - I trust your judgment about what we want/need to show and what we can safely leave out.

I also noticed we don't have an "Update" section for Set. In the context of Set, "Update" is either adding or removing elements, or _maybe_ querying for the set element and updating one of its properties. You may feel this is adequately covered elsewhere and that's why there's no "Update" example. But just wondering if that was an intentional decision, or just that there was no existing "update" section so we didn't add one. Up to you whether you think there's value in that.

cbullinger commented:
@dacharyc re-did this work to address the issues we discussed: 

- updated the code examples (split tests out to the Create, Read, etc. files like you did for dictionary) and snippets to hopefully provide better context 

- chose not to include an update example because there's an Update a Collection, which is how you'd update a set (i updated the copy to make that more explicit). adding another example didn't seem to provide much value, but i'm open to including it if it makes sense

- updated to-many relationship description to make it more explicit you can use realmSet

dacharyc reviewed (approved):
Overall, this looks good to me. Just a few small Qs and comments for your consideration, but nothing blocking.

cbullinger commented on examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/ReadTest.kt:
```diff
@@ -28,6 +29,60 @@ class ReadTest_Frog : RealmObject {
+
+                // Query for specific snacks
+                val wormsSnack = frog.favoriteSnacks.first { it.name == "Worms" }
+                assertEquals(3, frog.favoriteSnacks.size) // :remove:
```
Updated

cbullinger commented on examples/kotlin/shared/src/commonTest/kotlin/com/mongodb/realm/realmkmmapp/ReadTest.kt:
```diff
@@ -28,6 +29,60 @@ class ReadTest_Frog : RealmObject {
+                    }
+                )
+                // Query for frogs that have worms as a favorite snack
+                val frogs = query<RealmSet_Frog>("favoriteSnacks.name == $0", "Worms").find().first()
```
Updated
